asyncapi.yaml

```diff
@@ -0,0 +1,93 @@
+channels:
+  event-gateway-demo:
+    description: Demo Kafka topic for asyncapi-event-gateway-demo server. Users can send their events to this topic and see how message validation happens on the fly based on this right AsyncAPI file by connecting to `event-gateway-demo-validation-events` channel (`asyncapi-event-gateway-demo-validation` ws server).
+    publish:
+      message:
+        $ref: 'https://bit.ly/asyncapi#/components/messages/lightMeasured'
+  event-gateway-demo-validation-events:
+    description: Validation errors are published here, so users can see how message validation happens on the fly based on this right AsyncAPI file.
+    subscribe:
+      message:
+        $ref: '#/components/messages/ValidationError'
+  event-gateway-demo-validation:
+    description: Validation errors are published to and consumed from it. AsyncAPI Event-gateway is the only user of this channel. It can be consumed and exposed via `event-gateway-demo-validation-events` channel (`asyncapi-event-gateway-demo-validation` ws server).
+    subscribe:
+      message:
+        $ref: '#/components/messages/ValidationError'
+components:
+  messages:
+    ValidationError:
+      payload:
+        type: object
+        properties:
+          ts:
+            type: string
+            description: RFC-3339 date-time. Date and time when the message was validated.
+          errors:
+            type: array
+            description: Array of string. Validation errors.
+          msg:
+            $ref: '#/components/schemas/Message'
+      examples:
+        - payload:
+            ts: '2021-09-10T12:04:18:475203609Z'
+            errors: [ 'lumens: Invalid type. Expected: integer, given: string' ]
+            msg:
+              context:
+                channel: 'event-gateway-demo'
+              key: 'YXN5bmNhcGktd2FzLWhlcmU='
+              value: 'eyJsdW1lbnMiOiAid2hhdGV2ZXIifQ=='
+  schemas:
+    Message:
+      type: object
+      properties:
+        key:
+          type: string
+          description: Kafka message key (base64).
+        value:
+          type: string
+          description: Kafka message value (base64).
+        context:
+          $ref: '#/components/schemas/MessageContext'
+    MessageContext:
+      type: object
+      properties:
+        channel:
+          type: string
+          description: Name of the channel the message was published to.
+    ## the following schemas are needed due to a bug in parser-go: https://github.com/asyncapi/parser-go/issues/82
+    sentAt:
+      $ref: 'https://bit.ly/asyncapi#/components/schemas/sentAt'
+    lightMeasuredPayload:
+      $ref: 'https://bit.ly/asyncapi#/components/schemas/lightMeasuredPayload'
+  # the following messageTraits are needed due to a bug in parser-go: https://github.com/asyncapi/parser-go/issues/82
+  # $ref can't be used on messageTraits components, that's the reason it is hardcoded from http://bit.ly/asyncapi
```

Comment:
If there is a bug, so maybe we should copy needed content (whole, not only part as with `messageTraits`) from `https://bit.ly/asyncapi`? With

```yaml
  messageTraits:
    commonHeaders:
```

for me it's look strange 😆

Reply:
It looks strange indeed. I thought it was still worth in order to simplify the file but at the end there are so many modifications.
I'm gonna copy/paste whatever is needed and mention this is just based on the streetlights.yaml.

Reply:
It's a thing now.